window-copy.c

```diff
@@ -3097,26 +3118,40 @@ window_copy_search(struct window_mode_entry *wme, int direction, int regex,
 	wrapflag = options_get_number(wp->window->options, "wrap-search");
 	cis = window_copy_is_lowercase(str);
 
-	if (direction)
+	/*
+	 * Behave according to mode-keys. If it is emacs, search forward leaves the
+	 * cursor after the match. If it is vi, the cursor remains at the beginning of
+	 * the match, regardless of direction.
+	 */
+	keys = options_get_number(wp->window->options, "mode-keys");
+	if (direction) {
+		if (keys == MODEKEY_VI) {
+			if (data->foundlen <= 0)
+				window_copy_move_right(s, &fx, &fy, wrapflag);
+			else
+				for (i = 0; i < data->foundlen; i++)
```

Comment:
@nicm could you provide some guidance on this please? I'd like to only move all the way right of a match if the cursor is currently positioned at the beginning of the match.

I'm having a hard time wrapping my head around the various coordinate systems and data structures that tmux uses, so I don't know where to begin when it comes to figuring this out.